pkg/runtime/grpc/grpc-runtime.go

```diff
@@ -364,8 +397,44 @@ func (r *Runtime) runBuiltInGadgetOnTargets(
 
 func (r *Runtime) dialContext(dialCtx context.Context, target target, timeout time.Duration) (*grpc.ClientConn, error) {
 	opts := []grpc.DialOption{
-		grpc.WithTransportCredentials(insecure.NewCredentials()),
 		grpc.WithBlock(),
+		grpc.WithReturnConnectionError(),
+	}
+
+	if tlsKey, tlsCert, tlsCA := r.globalParams.Get(ParamTLSKey).String(), r.globalParams.Get(ParamTLSCert).String(), r.globalParams.Get(ParamTLSServerCA).String(); tlsKey != "" || tlsCert != "" || tlsCA != "" {
```

Comment:
I think this sets the new longest line record.

Reply:
So where is my trophy, then? 😆